galsim/table.py

```diff
@@ -149,19 +138,35 @@ def __init__(self, x, f, interpolant='spline', x_log=False, f_log=False):
         if self.f_log and np.any(self.f <= 0.):
             raise GalSimValueError("Cannot interpolate in log(f) when table contains f<=0.", f)
 
+        # Check equal-spaced arrays
+        if self._interp1d is not None:
+            if self.x_log:
+                logx = np.log(self.x)
+                dlogx = np.diff(np.log(self.x))
```

Comment:
First, it looks like logx isn't used.  I'm sure you intended to use it in the dlogx line.

But also, suspect it might be faster to use division rather than log for this:
```
ratio = self.x[1:] / self.x[:-1]
if not np.allclose(ratio, ratio[0]):